### ISSUE TITLE
Alignment for bullet / Numbered list 

### ISSUE THREAD

**Issue by vallikumar** [ckeditor/ckeditor4#3749]
Normally, a user can align the bullet lists/numbered lists and it should align left right or center properly along with the bulleted points or the numbers..

CK Editor is not supporting the align the bullet lists/numbered lists properly to left/right / center or justify. We have tried with CK editor version 4 and 5.

Steps to reproduce.
 1. Enter some random text e.g Test1 , Test2, Test3, Test4
2. Select the bullet list.
3. Now click on Aligned left, right , justify and center
4. Observe only text moves left to right but not bullet / numbered list. 

Expected result:
The bullet list / Numbered list should also move along with the text.
Actual Results:
But only the text is moving and the bullet points are not moving along.

Browser: …Chrome 78
OS: …Windows 10
CKEditor version: …7.1
Installed CKEditor plugins: …standard default plugins downloaded as part of package.

**Comment by jacekbogdanski** [comment]
I agree that it would look much better if the list item bullet (or any prefix like a number) is moved alongside with an item. We are using semantic HTML for content lists, but it seems like such issue could be easily fixed by setting:
```css
li {
  list-style-position: inside;
}
```
CSS style.

**Comment by Mykola-Veryha** [comment]
The settings didn't fix the problem. 
Any news how to fix the bug?

**Comment by KarolDawidziuk** [comment]
After F2F talk with @jacekbogdanski, we have some insights regarding to this ticket.

The simples solution mentioned by @jacekbogdanski to add extra style to the `li` element, will not work for block elements like headings.

There was a several attempts to handle it but it was not good, like creating a new pseudo-element `:before` in the first child of the list item and hiding the original `::marker`. After that way there would be another problem with calculate the appropriate value of the item. 

To deal with this we could create a some king of observer that will be watch list behaviours like removing positions from list and updating value.

And ever if we could handle this issue in this way, we would have problems with pseudo-elements in IE. So, this could possibly be changed to add some `span` to the list, however this would create further problems. 
 
For now this would be related to another plugins except `justify` and behaviours like: the big part of list logic and clipboard ( copy / paste ).

**Comment by jacekbogdanski** [comment]
Thanks @KarolDawidziuk for this research. Looking at the complexity of this feature it's unlikely we will be able to work on it in the nearest future as it would require rewriting a major part of the `list` plugin itself. 

For now, it's recommended to use styling from https://github.com/ckeditor/ckeditor4/issues/3749#issuecomment-568443955 as a workaround in specific applications.
